contracts/TokenManager.sol

```diff
@@ -229,14 +234,34 @@ contract TokenManager is
     }
 
     /**
-     * @notice  Burns the tokens and requests the closing of the vault
+     * @notice  Request the closing of the vault
+     * @dev     Sets the withdrawRequests mapping to the current timestamp
+     * @param   uuid  uuid of the vault/DLC
+     */
+    function requestCloseVault(bytes32 uuid) external whenNotPaused {
+        DLCLink.DLC memory dlc = dlcManager.getDLC(uuid);
+        if (dlc.creator != msg.sender) revert NotOwner();
+        withdrawRequests[uuid] = block.timestamp;
+    }
+
+    /**
+     * @notice  Burns the tokens and finishes the closing of the vault
+     * @dev     withdrawDelay has to have passed
      * @dev     User must have enough dlcBTC tokens to close the DLC fully
      * @param   uuid  uuid of the vault/DLC
      */
     function closeVault(bytes32 uuid) external whenNotPaused {
         DLCLink.DLC memory dlc = dlcManager.getDLC(uuid);
         if (dlc.creator != msg.sender) revert NotOwner();
 
+        if (withdrawRequests[uuid] == 0) revert NoWithdrawRequestMade();
+
+        if (withdrawRequests[uuid] + withdrawDelay > block.timestamp)
```

Comment:
you can use your isDelayPassed function here, rather than writing the logic twice